api/envoy/service/discovery/v3/discovery.proto

```diff
@@ -174,9 +179,19 @@ message DeltaDiscoveryRequest {
   // A list of Resource names to add to the list of tracked resources.
   repeated string resource_names_subscribe = 3;
 
+  // As with *resource_names_subscribe* but used when subscribing to resources indicated
+  // by a *udpa.core.v1.ResourceLocator*.
+  // [#not-implemented-hide:]
+  repeated udpa.core.v1.ResourceName udpa_resources_subscribe = 8;
```

Comment:
Shouldn't this be a `ResourceLocator`, not a `ResourceName`?

Reply:
In the design document it is a resource locator, but I'm wondering if that was a mistake. The client will only ever ask for a named resource, directives are resolved locally.

Reply:
I don't think that was a mistake.  The whole point of the division between `ResourceLocator` and `ResourceName` is that the former is to be used for matching semantics in requests and the latter is to be used to name concrete resources in responses.  Glob patterns need to be expressed as a `ResourceLocator`, not a `ResourceName`.  And if in the future we add more flexible matching for context params, that would be encoded in `ResourceLocator` as well.

It's true that `ResourceLocator` includes directives, but I think that's okay.  We went back and forth on this in the design, and I think where we wound up was that we would just not use the directives field of `ResourceLocator` in the context of `DeltaDiscoveryRequest`.  But I would also be okay with the other approach we had discussed of splitting up `ResourceLocator` into a base message without directives and a wrapper message that includes the base message and adds directives.  We could then use the base message here and the wrapper message in resources we send to the client.

Reply:
Right, thanks for the reminder. I've updated to make these locators.